lib/api/utils.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+// TODO: Tests
+const schemas = require('../database/schemas');
+const utils = require('../utils');
+
+const liveSchema = utils.singleItem('name', 'live', schemas).fields.map(field => {
+  return field.name;
+});
+
+/*
+ * Formats raw live data in to consumable returns
+ *
+ * @param {array} content - Array of objects representing the raw DB query of content
+ *
+ * @returns {array} - formatted object of content
+ */
+const format = (content, attrs) => {
+  return content.map(item => {
+    const result = {
+      id: item.id,
+      type: item.type,
+      type_slug: item['type-slug'], // eslint-disable-line camelcase
+      key: item.key,
+      key_slug: item['key-slug'], // eslint-disable-line camelcase
+    };
+
+    if (attrs) {
+      result.attributes = item.attributes;
+    }
+    else {
+      result.meta = {
+        url: `/api/types/${item['type-slug']}/${item.id}`,
+      };
+    }
+
+    return result;
+  });
+};
+
+/*
+ * Determines raw live data organization
+ *
+ * @param {object} params - Request parameters
+ *
+ * @returns {object} - Object with sort key, sort direction, pagination count and offset
+ */
+const organize = (params, lookup) => {
+  let sort = params.sort || 'key';
+  let sortDir = params.sort_dir || 'asc';
+  let sortKeys = lookup || liveSchema;
+
+  let page = parseInt(params.page, 10) || 1;
+  let perPage = parseInt(params.per_page, 10) || 30;
+  let count = 0;
+
+
+  //////////////////////////////
+  // Sorting
+  //////////////////////////////
+  // Ensure sort is an available key
+  sortKeys = sortKeys.filter(key => {
+    if (key === 'attributes') {
+      return false;
+    }
+
+    return true;
+  });
+  if (sortKeys.indexOf(sort) < 0) {
+    if (lookup) {
+      sort = lookup[0];
+    }
+    else {
+      sort = 'key';
+    }
+  }
+
+  // Ensure sorting is either ascending or descending
+  if (sortDir !== 'asc' && sortDir !== 'desc') {
+    sortDir = 'asc';
+  }
+
+  //////////////////////////////
+  // Pagination
+  //////////////////////////////
+  // Ensure there is at least one page returned
+  if (page < 1) {
+    page = 1;
+  }
+
+  // Ensure pagination fits within parameters
+  if (perPage < 1) {
+    perPage = 1;
+  }
+  else if (perPage > 100) {
+    perPage = 100;
```

Comment:
Yes, we max out at 100 items per page. It's the limit GitHub imposes, so I thought that was good enough for our stuff. In the Gherkin